provisioning/device/samples/register_x509.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var fs = require('fs');
+var Http = require('azure-iot-provisioning-device-http').Http;
+var X509Security = require('azure-iot-security-x509').X509Security;
+var provisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+var idScope = '[id scope]';
+var registrationId = '[registration id'];
+var deviceCert = {
+  cert: fs.readFileSync('[cert filename]'.toString()),
+  key: fs.readFileSync('[key filename]'.toString())
+};
+
+var transport = new Http(idScope);
+var securityClient = new X509Security(deviceCert);
+var deviceClient = provisioningDeviceClient.create(transport, securityClient);
+
+deviceClient.register(registrationId, false, function(err, result) {
```

Comment:
>register [](start = 13, length = 8)

consider adding a comment explaining the arguments used (especially "false" in the middle here with no context is a bit cryptic - I wonder if we should make it optional)